src/GafferSceneUI/Inspector.cpp

```diff
@@ -296,6 +297,8 @@ void Inspector::inspectHistoryWalk( const GafferScene::SceneAlgo::History *histo
 						{
 							result->m_editFunction = nonEditableReason;
 						}
+
+						result->m_disableEditFunction = disableEditFunction( result->m_source.get(), history );
```

Comment:
I think we're making the right decision in not allowing edits to be disabled unless they are in the current EditScope. The guarantee that the UI won't touch anything outside of the scope is a fundamental one.

But at present it's not obvious why something can't be disabled when it's outside the scope - it took me a little while to figure out in my first interactive testing for this PR, even though I was familiar with the rules from before. At the moment, `m_disableEditFunction` remains uninitialised in this case, but perhaps we can initialise it with a message describing the reason it can't be disabled? I'm pretty sure we guarantee that `m_editFunction` is always initialised, and it seems like the same might be good for `m_disableEditFunction`.

Reply:
I've taken a stab at describing as many of the non-disablable reasons as part of the overall messaging improvements in 1968cbf. Some of the wording could potentially be improved, suggestions very welcome there, I've tried to keep things as terse as possible given the ephemeral nature of the popups...